### ISSUE TITLE
Clarify behavior of combination of SetVirtualHostNameToFolderMapping and WebResourceRequestFilter

### ISSUE THREAD

**Issue by Urmeli0815** [MicrosoftEdge/WebView2Feedback#2103]
Hi,

I'm currently trying the following with my WebView:
- first I call AddWebResourceRequestedFilter to install a filter for all resources
- then I add a WebResourceRequested event handler
- finally I call SetVirtualHostNameToFolderMapping to provide a fall-through for all requests that weren't handled by my event handler. I thought this should work because in one of the samples it is mentioned that "If put_Response is not called, the request will continue as normal."

When I try it like that my WebResourceRequested event handler is never called. It seems that SetVirtualHostNameToFolderMapping grabs the request before. The documentation of SetVirtualHostNameToFolderMapping doesn't contain anything obvious that describes the behavior. So it seems these two functionalities can't be combined.

Can you clarify this and maybe also add some info in the docs?

Thank you in advance!

**Comment by champnic** [comment]
I believe what's happening here is that `SetVirtualHost...` is turning the web request into a file open in the set folder, so it never goes through the `WebResourceRequested` event.

Can you share the relevant setup code?

@LiangTheDev may also have ideas about what's going on here.

**Comment by LiangTheDev** [comment]
That is correct. We should update our doc to state that WebResourceRequested event will not be fired for the requests to the mapped virtual hosts.

**Comment by champnic** [comment]
I've changed this to a doc bug - we'll get the information added shortly!

**Comment by wall-git** [comment]
> 
Are there other ways to do this without building our own Web Server?
[https://github.com/MicrosoftEdge/WebView2Feedback/issues/2685](url)

**Comment by LiangTheDev** [comment]
If you want to provide different response during runtime for different urls, then you would have to use WebResourceRequested handler for all requests to those virtual host.

**Comment by fmongeau** [comment]
@champnic Due to some requirements on our end, our web application will be hosted either with nginx or in a webview2 in a windows application. When hosted with nginx, we forward data requests to our backend. So our web pages use relative paths to access data. To host that same web application in webview2, I'd like to virtual host the webpages and still intercept web requests for the same domain with the WebResourceRequested event.  Since the same web pages are hosted in 2 environments, I'd prefer not to have to adapt all the pages to specify a different domain.

`
webView.CoreWebView2.SetVirtualHostNameToFolderMapping("test.fred", @"c:\dev\website1", Microsoft.Web.WebView2.Core.CoreWebView2HostResourceAccessKind.Allow);
webView.CoreWebView2.AddWebResourceRequestedFilter("http://test.fred/api/*", CoreWebView2WebResourceContext.All);
webView.CoreWebView2.WebResourceRequested += CoreWebView2_WebResourceRequested;
`

You seem to indicate that this issue is a documentation issue and that the documentation should indicate that requests to the virtual host name cannot be intercepted with the WebResourceRequested event. Makes me wonder if it could be. If a request resource does not exist in the folder that is virtually hosted, could it fallback on the WebResourceRequested event ?

Thanks

**Comment by champnic** [comment]
Hey @fmongeau - the documentation issue is just to clarify the current behavior. If you like you can add a feature request to change the behavior and make it easier to use both virtual host names and WebResourceRequests together.

**Comment by fmongeau** [comment]
> Hey @fmongeau - the documentation issue is just to clarify the current behavior. If you like you can add a feature request to change the behavior and make it easier to use both virtual host names and WebResourceRequests together.

Will do, thanks @champnic
